tests/test_bootpolicy.robot

```diff
@@ -0,0 +1,187 @@
+    Response Should Be Equal   CDROM
+
+    Power On Host
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   PERMANENT
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   CDROM
+
+Onetime boot order after warm reset
+    [Documentation]   This testcase is to verify that boot policy and order does not change  
+    ...               after warm reset on a system with onetime boot policy.
+
+    Power On Host
+
+    ${bootpolicy} =   Set Variable   ONETIME
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   ONETIME
+
+    Trigger Warm Reset
+ 
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   ONETIME
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   Network
+ 
+Permanent boot order after warm reset
+    [Documentation]   This testcase is to verify that boot policy and order does not change  
+    ...               after warm reset on a system with permanent boot policy.
+
+    Power On Host
+
+    ${bootpolicy} =   Set Variable   PERMANENT
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   PERMANENT
+
+    ${bootDevice} =   Set Variable   CDROM
+    ${valueDict} =   create dictionary   data=${bootDevice}
+    Write Attribute    /org/openbmc/settings/host0   boot_flags   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   CDROM
+
+    Trigger Warm Reset
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   PERMANENT
+
+    Read the Attribute  /org/openbmc/settings/host0    boot_flags
+    Response Should Be Equal   CDROM
+ 
+Set boot policy to invalid value
+    [Documentation]   This testcase is to verify that proper error message is prompted  
+    ...               when invalid value to provided to boot policy.
+   
+    ${bootpolicy} =   Set Variable   abc
+    ${valueDict} =   create dictionary   data=${bootpolicy}
+    Write Attribute    /org/openbmc/settings/host0   boot_policy   data=${valueDict}
+    Read the Attribute  /org/openbmc/settings/host0    boot_policy
+    Response Should Be Equal   abc
+    
+*** Keywords ***
+
+Response Should Be Equal
+    [arguments]    ${args}
+    Should Be Equal    ${OUTPUT}    ${args}
+
+Read the Attribute     
+    [arguments]    ${uri}    ${parm}
+    ${output} =     Read Attribute      ${uri}    ${parm}
+    set test variable    ${OUTPUT}     ${output}
+
+Power On Host
```

Comment:
Move Power On Host/Power Off Host/Warm Reset to lib/rest_client.robot..So that it can be used by other testcases.